CMakeLists.txt

```diff
@@ -145,6 +145,12 @@ else()
   include_directories(SYSTEM ${MLIR_INCLUDE_DIR})
   include_directories(SYSTEM ${MLIR_TABLEGEN_OUTPUT_DIR})
 
+  # If building as part of a unified build, whether or not MLIR's execution engine
+  # is enabled must be fetched from its subdirectory scope.
+  get_directory_property(MLIR_ENABLE_EXECUTION_ENGINE
+      DIRECTORY ${MLIR_MAIN_SRC_DIR}
+      DEFINITION MLIR_ENABLE_EXECUTION_ENGINE)
+
```

Comment:
This looks even cleaner as before, nice! It is also done in the separate build now, right? Did you test that it still works fine with that; because the JIT tests are only run with the unified build in the CI?

Reply:
If you look closely, all this code is in a big if statement that is only ran in the unified build. In any case, I tested both unified and separate, and they both get the JIT enabled properly.

Reply:
Ah, perfect!